src/CommonITILObject.php

```diff
@@ -4359,6 +4359,22 @@ public function getSearchOptionsSolution()
             ]
         ];
 
+        if ($itemtype === Ticket::class || $itemtype === Change::class || $itemtype === Problem::class) {
```

Comment:
This condition should be removed.

```suggestion
```

Reply:
Along with removing those conditions, you can also drop all the `get_class($this)` you added.